javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -213,6 +213,17 @@ object Character {
     else -1
   }
 
+  // ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/Character.java
+  def forDigit(digit: Int, radix: Int): Char = {
+    if ( radix < MIN_RADIX || radix > MAX_RADIX || digit < 0 || digit >= radix) {
+      0
+    }
+    else {
+      val overBaseTen = digit - 10
+      (if (overBaseTen < 0) '0' + digit else 'a' + overBaseTen).toChar
```

Comment:
You are right. Performance of your code is better. Lets leave it.
However, could you assign the result of the inner if to another val? parenthesized ifs are hard to read (and we don't have any of them AFAIK).